src/twisted/conch/ssh/keys.py

```diff
@@ -240,6 +301,12 @@ def _fromString_PRIVATE_BLOB(cls, blob):
         elif keyType == b'ssh-dss':
             p, q, g, y, x, rest = common.getMP(rest, 5)
             return cls._fromDSAComponents(y=y, g=g, p=p, q=q, x=x)
+        elif keyType in [b'ecdsa-sha2-' + curve for curve in list(_curveTable.keys())]:
+            x, y, private_value, rest = common.getMP(rest, 3)
+            newKey = cls._fromECComponents(x=x, y=y, curve=keyType,
+                private_value=private_value)
+            newKey.ecKeyName = keyType
```

Comment:
Might it be a good idea to include `ecKeyName` in the arguments to `_fromECComponents`?  It seems like the construction is incomplete without it.